### DISCUSSION TITLE
Voltage adjustment on vBIOS flashed gpu not showing

### DISCUSSION THREAD

**Discussion by frostyllama** [seerge/g-helper#2690] [Q&A]
Have an nvidia 4070 max-q at 60w flashed to 90w

The average wattage is higher but the GPU is still stuck with the 60w cap. The bios should allow for base voltage adjustment but the option doesn't appear in ghelper. The boost power slider doesn't have any effect, nor does the gpu clock offset. Memory clock offset is not 1 - 1, 500mhz results in about a 125mhz boost.

following command shows 90 watts:
nvidia-smi --query-gpu=power.max_limit --format csv,noheader,nounits

Asus services are all stopped. Msi afterburner shows a power lock as well with the option enabled. Tried DDU'ing drivers and installing recent stripped gpu drivers.

Any suggestions?

**Comment by seerge** [comment]
@frostyllama what kind of device do you use ? On everything but latest 2024 G14/G16 GPU TDP is locked somewhere in Nvidia drivers / BIOS level :) That's normal.

**Reply by frostyllama** [reply]
I've got a 2023 x13 4070 max-q 60w, but flashed it with the 2024 g14 4070 90w vbios which should be adjustable. The flash set a higher base clock range from a ~1350mhz cap up to ~1600mhz now.  Base wattage is higher but still capped at 60w. Reinstalled drivers after vbios flash. Is there a hardware lock of some kind?

**Reply by seerge** [reply]
There is a BIOS / driver lock as I mentioned above

**Comment by IceStormNG** [comment]
ASUS laptops have power limits enforced by the EC (Embedded controller). Even if you flash a higher wattage vBIOS, the firmware and EC will limit the GPU power to what ASUS set for the respective move (different limits for silent, performance and turbo).

**Reply by frostyllama** [reply]
Appreciate you mentioning the EC limits and profiles. With the g14 4070 firmware in turbo I was just able to get to 80w with nvidia-smi, super close to the 90w peak. Watched the gpu wattage with hwinfo. Got frame/power drops every 20 seconds or so at peak. Suspect it was hitting power draw limits (brick only 130w) and CPU stalls with the set profiles. With limiters I was able to maintain about a 70% FPS gain from stock in furmark with a 75c gpu temp limit.

I've read other Asus laptop vBIOS flashes get higher than specced wattages. Guess they're still influenced by the EC and profiles as you mentioned.  The G-Helper GPU dynamic boost doesn't do anything with the new vBIOS. Memory clock offset cut by 1/4. GPU clock/offset work within the g14 bios clock range.
